### PR TITLE
feat: support reload cli

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes [EMQX-10117](https://emqx.atlassian.net/browse/EMQX-10117).
1. Add `emqx_ctl conf reload`  to reload `etc/emqx.conf` on local node.
2. Check all configs' schema before load them.
3. Add more test for emqx_conf_cli.

### <samp>🤖 Generated by Copilot at 4006afe</samp>

This pull request improves the config management of EMQ X by adding a `conf reload` command, fixing a bug in the authorization source update, and adding a test case for the `emqx_conf_cli` module. It also introduces some code refactoring and enhancement, such as using a `warning/1` function, a `default_authz/0` function, and a `READONLY_KEYS` macro.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_conf/src/emqx_conf_cli.erl [hunk 1]
```diff
@@ -42,10 +46,16 @@ conf(["show"]) ->
     print_hocon(get_config());
 conf(["show", Key]) ->
     print_hocon(get_config(Key));
+conf(["load", "--auth-chains", AuthChains, Path]) when
+    AuthChains =:= "replace"; AuthChains =:= "merge"
+->
+    load_config(Path, AuthChains);
 conf(["load", Path]) ->
-    load_config(Path);
+    load_config(Path, "replace");
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf_cli.erl:54]: default should be replace or merge ?

File: apps/emqx_authz/src/emqx_authz.erl [hunk 2]
```diff
@@ -128,6 +124,9 @@ lookup(Type) ->
     {Source, _Front, _Rear} = take(Type),
     Source.
 
+merge(NewConf) ->
+    emqx_authz_utils:update_config(?ROOT_KEY, {?CMD_MERGE, NewConf}).
```
Comment (zhongwencool) [inline apps/emqx_authz/src/emqx_authz.erl:128]: remove authz from etc/emqx.conf and check schema default.

File: apps/emqx_authz/src/emqx_authz.erl [hunk 3]
```diff
@@ -158,18 +157,26 @@ pre_config_update(Path, Cmd, Sources) ->
...
     do_pre_config_update(Cmd, Sources);
+do_pre_config_update(?ROOT_KEY, {?CMD_MERGE, NewConf}, OldConf) ->
+    do_pre_config_merge(NewConf, OldConf);
 do_pre_config_update(?ROOT_KEY, NewConf, OldConf) ->
     do_pre_config_replace(NewConf, OldConf).
 
+do_pre_config_merge(NewConf, OldConf) ->
+    MergeConf0 = emqx_utils_maps:deep_merge(OldConf, NewConf),
+    NewSources = merge_sources(NewConf, OldConf),
+    do_pre_config_replace(MergeConf0#{<<"sources">> := NewSources}, OldConf).
```
[the remark this answers is not in the document]
Comment (zhongwencool) [reply apps/emqx_authz/src/emqx_authz.erl:168]: merge_sources(Old,New).

File: apps/emqx_authz/src/emqx_authz.erl [hunk 4]
```diff
@@ -628,3 +635,78 @@ check_acl_file_rules(Path, Rules) ->
...
+                        MergeSource = emqx_utils_maps:deep_merge(Old, New),
+                        {[MergeSource | OriginAcc], NewAcc1}
+                end
+            end,
+            {[], NewSources0},
+            OriginSources0
+        ),
+    lists:reverse(OriginSource) ++ NewSources.
+
+search(Type, Sources) ->
```
Comment (zhongwencool) [inline apps/emqx_authz/src/emqx_authz.erl:659]: search => take

File: apps/emqx_conf/src/emqx_conf_cli.erl [hunk 5]
```diff
@@ -164,22 +176,78 @@ hidden_roots() ->
...
-                end,
-                Conf
-            );
+        {ok, RawConf} when RawConf =:= #{} ->
+            emqx_ctl:warning("load ~ts is empty~n", [Path]),
+            {error, empty_hocon_file};
+        {ok, RawConf} ->
+            case check_config(RawConf) of
+                ok ->
+                    maps:foreach(fun(K, V) -> update_config(K, V, AuthChain) end, RawConf);
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf_cli.erl:193]: lists:keysort

File: apps/emqx_conf/src/emqx_conf.erl [hunk 6]
```diff
@@ -213,10 +216,119 @@ schema_module() ->
...
+    case check_config(SchemaMod, RawDefault) of
+        {ok, CheckedConf} ->
+            case
+                lists:filtermap(fun(Key) -> filter_changed(Key, CheckedConf) end, ?READONLY_KEYS)
+            of
+                [] ->
+                    {ok, maps:without([atom_to_binary(K) || K <- ?READONLY_KEYS], Raw)};
+                Error ->
+                    ?SLOG(error, #{
+                        msg => "failed_to_change_read_only_key_in_etc_config",
```
Comment (zhongwencool) [inline apps/emqx_conf/src/emqx_conf.erl:293]: update_readonly_keys_prohibited

**Review by HJianBo — APPROVED** [review]

**Pull request merged**